db/document.go

```diff
@@ -332,6 +333,16 @@ func (doc *Document) GetMetaMap(userXattrKey string) (map[string]interface{}, er
 	}, nil
 }
 
+func (doc *Document) SetCrc32cUserXattrHash(userXattrKey string) {
+	// If user xattr is nil or feature is disabled then set hash to ""
+	if len(doc.rawUserXattr) == 0 || userXattrKey == "" {
```

Comment:
Is there a scenario where userXattrKey == "", but len(doc.rawUserXattr) > 0?  Want to understand the requirement to pass userXattrKey in here.

Reply:
I think I've made a mistake when looking at Bens comment above but I think I need to change this back to what it was. I will re-review this.

Reply:
If the key is empty, logically the doc doesn't have a user xattr any more (because we don't know where to find one). I think it makes sense to empty the checksum in that case?

Reply:
Yeah. Ben is right. 
> userXattrKey == "", but len(doc.rawUserXattr) > 0
That bit threw me a little. But yeah that basically checks if the feature is enabled. Its feasible that a customer will enable / disable this and in that case we want to remove any previous uses of the hash.

Reply:
As discussed, I don't think there's a case where rawUserXattr is non-zero if the userXattrKey is empty.